gui/src/renderer/components/Switch.tsx

```diff
@@ -114,13 +115,19 @@ export default class Switch extends React.PureComponent<IProps, IState> {
     }
   };
 
-  private handleClick = () => {
+  private handleClick = async () => {
     if (this.props.disabled) {
       return;
     }
 
     if (!this.changedDuringPan) {
-      this.setState((state) => ({ isOn: !state.isOn, notifyOnTransitionEnd: true }));
+      const newIsOn = !this.state.isOn;
+
+      if (this.props.confirmation !== undefined && !(await this.props.confirmation(newIsOn))) {
+        return;
+      }
+
+      this.setState({ isOn: newIsOn, notifyOnTransitionEnd: true });
```

Comment:
If I may, updating state based on the current state should be done with a function.

```suggestion
      this.setState((prevState) => ({
        isOn: !prevState.isOn,
        notifyOnTransitionEnd: true,
      }));
```

Reply:
In this situation the variable `newIsOn` is also passed to an asynchronous callback for confirmation so the new state must be the value that has been confirmed. The only current use of this within our app is to show a confirmation dialog to the user.